File: go/glob.go
// testing globs against using gobwas/glob
package main

import (
	"fmt"
	"os"

	"github.com/gobwas/glob"
)

func main() {
	if len(os.Args) < 3 {
		fmt.Printf("Usage: %s <glob> <filename>\n", os.Args[0])
		os.Exit(1)
	}

	globExpr := os.Args[1]
	filename := os.Args[2]

	g, err := glob.Compile(globExpr)
	if err != nil {
		fmt.Printf("Error compiling %q: %s\n", globExpr, err)
		os.Exit(1)
	}

	if g.Match(filename) {
		fmt.Printf("%q matches %q\n", filename, globExpr)
	} else {
		fmt.Printf("%q does not match %q\n", filename, globExpr)
	}
}

File: ruby/glob.rb
# testing against builtin glob library

if ARGV.size < 2
  warn "Usage: #{$0} <glob> <filename>"
  exit 1
end

glob = ARGV[0]
filename = ARGV[1]

begin
  if File.fnmatch(glob, filename, File::FNM_EXTGLOB | File::FNM_PATHNAME)
    puts "#{filename.inspect} matches #{glob.inspect}"
  else
    puts "#{filename.inspect} does not match #{glob.inspect}"
  end
rescue => e
  warn "Problem testing #{filename.inspect} against #{glob.inspect}"
  warn e
  exit 1
end
